src/App.vue

```diff
@@ -1,55 +1,159 @@
 <script lang="ts">
 import { defineComponent } from 'vue';
 import Main from './components/Main.vue'
+import MobileMenu from './components/MobileMenu.vue'
+import CartFlyout from './components/CartFlyout.vue';
 
 export default defineComponent({
   name: 'App',
+  components: {
+    Main,
+    MobileMenu,
+    CartFlyout
+  },
+  data(){
+    return {
+      mobileMenuOpen: false,
+      quantity: 0,
+      cartOpen: false,
+    }
+  },
+  methods: {
```

Comment:
You can also use interfaces to define your component's methods instead of defining them as a plain object.
```
interface AppMethods {
  increment: () => void;
  decrement: () => void;
}
```